src/aegis_ai/tools/cwe/__init__.py

```diff
@@ -4,34 +4,43 @@
 import json
 import logging
 import os
+from typing import List, Dict, Any
 
 from zipfile import ZipFile
 
 import requests
 from pydantic import Field
 from pydantic_ai import Tool, RunContext
+from pydantic_ai.toolsets import FunctionToolset
 
 from aegis_ai import config_dir
 from aegis_ai.data_models import CWEID, cweid_validator
 from aegis_ai.tools import BaseToolOutput, default_tool_http_headers, BaseToolInput
 
 logger = logging.getLogger(__name__)
 
-# retrieve from cwe.mitre.org
+JsonBlob = Dict[str, Any]
+
+# retrieve allowed cwes from cwe.mitre.org
 CWE_URLS = [
     "https://cwe.mitre.org/data/csv/699.csv.zip",  # development - the only view supported by OSIM
     # "https://cwe.mitre.org/data/csv/1000.csv.zip",  # research
     # "https://cwe.mitre.org/data/csv/1008.csv.zip",  # architectural
     # "https://cwe.mitre.org/data/csv/1081.csv.zip",  # entries with maintenance notes
 ]
 
+# lazy load CWE
+CACHE_DIR = f"{config_dir}/mitre_cwe"
+CACHE_FILE = "cwe_full_defs.json"
+os.makedirs(CACHE_DIR, exist_ok=True)
```

Comment:
@JimFuller-RedHat Do we really want to create directories on module import?  That feels unexpected to me.

Reply:
happy to  move the call to whenever we persist the cache - makes sense.